oaknational/oak-ai-lesson-assistant
# feat: use TRPC for state and remove the spinner flash

## Description

- If the server has any kind of failure after a message has been sent, then it will fail to send the "state" message
- This results in some odd behaviour on the front end
- We now have TRPC for loading the initial chat, so... let's get rid of the state command use TRPC to fetch the chat after each message has arrived
- This moves the TRPC call inside the ChatProvider, which can then refetch the query
- For performance we change the tempLessonPlan to use a ref instead of state, which improves performance during streaming
- Plus, by doing so we have more control and can get rid of the flash where the lessonPlan is replaced by the new state, which currently makes each section flip back to the loading state
- The end result is that the front end doesn't jump and show spinners after each message, plus we can guarantee after each message that the front end has a matching lesson plan state to what the server has

## How to test

1. Go through the chat flow
2. Keep an eye on the sections as they appear on the right
3. You should not see the 500ms spinner animation after the message to the user appears on the left
4. Reloading the chat you should see the same state
5. In the Network panel you should see TRPC calls to fetch the chat after each message has completed
